api/v1alpha1/stardogrole_types.go

```diff
@@ -25,19 +26,44 @@ import (
 
 // StardogRoleSpec defines the desired state of StardogRole
 type StardogRoleSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+	// RoleName describes the name of a Role.
+	RoleName string `json:"roleName,omitempty"`
+	// StardogInstanceRef references the StardogInstance object in which the role is maintained.
+	StardogInstanceRef string `json:"stardogInstanceRef,omitempty"`
+	// Permissions lists the permissions assigned to a role
+	Permissions []StardogPermissionSpec `json:"permissions,omitempty"`
+}
 
-	// Foo is an example field of StardogRole. Edit StardogRole_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+// StardogPermissionSpec defines a Stardog permission assigned to a Role
+type StardogPermissionSpec struct {
+	// Action describes the action a specific permission is assigned to
+	// Types of Actions: ALL, CREATE, DELETE, READ, WRITE, GRANT, REVOKE, EXECUTE
+	Action string `json:"action,omitempty"`
```

Comment:
Please indicate whether the actions support listing (e.g. comma or space separated? ) and whether they're uppercase, lowercase or case insensitive.
Is this `// Types of Actions` a special usage of comments, e.g. does OperatorSDK create an enum out of this? or is this just a regular comment?
If it's not possible to provide multiple actions, we can make it so that the K8s API will reject invalid values:
Try if that works:
```suggestion
	// Action describes the action a specific permission is assigned to
	// +kubebuilder:validation:Enum=All;Create;Delete;Read;Write;Grant;Revoke;Execute
	Action string `json:"action,omitempty"`
```
According to the [examples](https://book.kubebuilder.io/reference/generating-crd.html) it uses a new type for that:
```go
        Action ActionType `json:"action,omitempty"`
...

// +kubebuilder:validation:Enum=All;Create;Delete;Read;Write;Grant;Revoke;Execute
type ActionType string
```

Reply:
In the example you provide it seems we don't need a new type that ultimately is string type.